### PR TITLE
Handle errors in delete worker

### PR DESCRIPTION
**Pull request by craiga**
* Set error state in state manager.
* Remove message from queue.
* Reflect the structure of the create worker in the delete worker.

Resolves issues found during testing of #422.

giphy-16

### CODE REVIEW

File: medicines/doc-index-updater/src/create_manager/mod.rs [hunk 1]
```diff
@@ -56,13 +56,15 @@ async fn try_process_from_queue(
...
-                    );
+                    let _ = state_manager
+                        .set_status(
+                            retrieval.message.job_id,
+                            JobStatus::Error {
+                                message: "Couldn't find file".to_string(),
+                                code: "404".to_string(),
+                            },
+                        )
+                        .await?;
```
Comment (craiga) [inline medicines/doc-index-updater/src/create_manager/mod.rs]: I found that `.await?` was required to set the status in the delete process.

Comment (TimboTambo) [reply medicines/doc-index-updater/src/create_manager/mod.rs]: Ah yeah - noticed this yesterday when I was looking into and completely forgot to do anything with it!

File: medicines/doc-index-updater/src/models.rs [hunk 2]
```diff
@@ -29,11 +30,18 @@ impl FromStr for JobStatus {
...
-                code: "0x0".to_owned(),
-            }),
-            e => Err(format!("Status unknown: {}", e)),
+            status => {
+                // If this message is in the format "Error(error code: error message)",
+                // reconstruct it into JobStatus::Error.
+                let error_re = Regex::new(r"^Error\((?P<code>[^:]*): (?P<message>.*)\)$").unwrap();
+                match error_re.captures(status) {
+                    Some(capture) => Ok(JobStatus::Error {
+                        message: capture.name("message").unwrap().as_str().to_string(),
```
Comment (TimboTambo) [inline medicines/doc-index-updater/src/models.rs]: These go through quite a journey! A result -> str -> String!

Comment (craiga) [reply medicines/doc-index-updater/src/models.rs]: It's nuts. The captures have an `as_str` but no `to_string`.

**Pull request merged**